sdk/component/store/local/store.go

```diff
@@ -171,46 +199,90 @@ func (m *manager) Write(ctx context.Context, instanceID uuid.UUID, findings []*o
 
 // Update updates existing vulnerabilities in the underlying database.
 // It returns ErrNoFindingsFound if the passed instanceID is not found.
-func (m *manager) Update(ctx context.Context, instanceID uuid.UUID, findings []*ocsf.VulnerabilityFinding) error {
-	jsonFindings, err := m.marshalFindings(findings)
+func (m *manager) Update(ctx context.Context, instanceID uuid.UUID, findings []*vf.VulnerabilityFinding) error {
+	tx, err := m.db.BeginTx(ctx, nil)
 	if err != nil {
-		return err
+		return errors.Errorf("could not start update transaction: %w", err)
 	}
 
-	stmt, err := m.db.PrepareContext(ctx, `
+	rollback := func(tx *sql.Tx, err error) error {
+		if txErr := tx.Rollback(); txErr != nil {
+			return errors.Errorf("could not rollback transaction for error %w: %w", err, txErr)
+		}
+		return errors.Errorf("unexpected update error, rolled back: %w", err)
+	}
+
+	defer func() {
+		if err := tx.Rollback(); err != nil {
+			// TODO: replace with logger.
+			log.Printf("failed to rollback update transaction: %s", err)
+		}
+	}()
+
+	for _, finding := range findings {
+		b, err := protojson.Marshal(finding.Finding)
+		if err != nil {
+			return rollback(
+				tx,
+				errors.Errorf("could not marshal JSON finding: %w", err),
+			)
+		}
+
+		stmt, err := tx.PrepareContext(ctx, `
 		UPDATE finding 
 			SET 
-			    findings = :findings,
+			    details = :details,
 			    updated_at = :updated_at
-			WHERE 
+			WHERE
+			    id = :id AND
 			    instance_id = :instance_id
 		;
-	`)
-	if err != nil {
-		return errors.Errorf("could not prepare update statement: %w", err)
-	}
+		`)
+		if err != nil {
+			return rollback(
+				tx,
+				errors.Errorf("could not prepare update statement: %w", err),
+			)
+		}
 
-	defer stmt.Close()
+		defer stmt.Close()
```

Comment:
we are not preserving over here the error that could potentially be generated

Reply:
Fixed